mdp/nodes/recursive_expansion_nodes.py

```diff
@@ -0,0 +1,623 @@
+
+def init_legendre_rational(result, x, pos, cur_var):
+    """Initialize the first and second order before starting the recursion.
+
+    Mathematically this amounts to
+    R_1 = (x-1)/(x+1)
+    R_2 = 3/2 * (x-1)/(x+1)*P_1 - 1/2.
+
+    :param result: Contains the observations along the first dimension
+        and the different function values of expansion w.r.t. an observation
+        along the second dimension.
+    :type result: numpy.ndarray
+
+    :param x: The data to be expanded.
+    :type x: numpy.ndarray
+
+    :param pos: The index of the element to be computed, along the second
+        dimension of result.
+    :type int: int
+
+    :param cur_var: The index of the current variable to be considered in the
+        recursion. This value will have to be lower than x.shape[1].
+    :type cur_var: int
+
+    :returns: The index of the next element that will be computed in the
+        recursion, the order of the next element, and the index of a special
+        member.
+    :rtype: numpy.ndarray
+    """
+    result[:, pos] = (x[:, cur_var]-1.)/(x[:, cur_var]+1.)
+    # after the first variable the second order is needed (no zero order available)
+    result[:, pos+1] = 3./2.*(x[:, cur_var]-1.) / (x[:, cur_var]+1.) \
+        * result[:, pos] - 1. / 2.
+    return pos+2, 3, None
+
+
+def recf_legendre_rational(result, x, special, n, cur_var, pos):
+    """Implementation of the recursion formula for Legendre rational functions.
+    The recursion formula is
+    R_n = (2n-1)/n * (x-1)/(x+1) R_{n-1} - (n-1)/n R_{n-2}.
+
+    :param result: Contains the observations along the first dimension
+        and the different function values of expansion w.r.t. an observation
+        along the second dimension.
+    :type result: numpy.ndarray
+
+    :param x: The data to be expanded.
+    :type x: numpy.ndarray
+
+    :param special: Index of a special element to be considered in the
+        recursion formula. E.g. the first order function value.
+    :type special: int
+
+    :param n: The order of the function to be computed at this step.
+    :type n: int
+
+    :param cur_var: The index of the current variable to be considered in the
+        recursion. This value will have to be lower than x.shape[1].
+    :type cur_var: int
+
+    :param pos: The index of the element to be computed, along the second
+        dimension of result.
+    :type int: int
+
+    :returns: The vectorized result (along the observations) of the
+        n-th recursion step of the cur_var-th variable.
+    :rtype: numpy.ndarray
+
+    .. admonition:: Reference
+
+        https://en.wikipedia.org/wiki/Legendre_rational_functions
+    """
+    return (2.*n-1.)/n*(x[:, cur_var]-1.) / (x[:, cur_var]+1.) \
+        * result[:, pos-1] - (n-1.) / n * result[:, pos-2]
+
+
+def init_chebyshev_poly(result, x, pos, cur_var):
+    """Initialize the first and second order before starting the recursion.
+
+    Mathematically this amounts to
+    T_1 = x
+    T_2 = 2xT_1 -1.
```

Comment:
For better readablility write it 2x T_1 - 1 or even 2x * T_1 - 1. Omitting * when multi-character variable names are involved should be considered carefully.

Reply:
Fixed.